application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksPaginatorInteractor.java

```diff
@@ -0,0 +1,483 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.emoji.Emoji;
+import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.component.SelectMenuInteractionEvent;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.ComponentInteraction;
+import net.dv8tion.jda.api.interactions.components.LayoutComponent;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import net.dv8tion.jda.api.interactions.components.selections.SelectMenu;
+import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
+
+import org.togetherjava.tjbot.commands.UserInteractionType;
+import org.togetherjava.tjbot.commands.UserInteractor;
+import org.togetherjava.tjbot.commands.componentids.ComponentIdGenerator;
+import org.togetherjava.tjbot.commands.componentids.ComponentIdInteractor;
+import org.togetherjava.tjbot.commands.utils.Arraylizable;
+import org.togetherjava.tjbot.db.generated.tables.records.BookmarksRecord;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+
+/**
+ * This class can create a paginated message for listing or removing bookmarks. A paginated message
+ * is requested by calling
+ * {@link BookmarksSystem#requestListPagination(GenericCommandInteractionEvent)} or
+ * {@link BookmarksSystem#requestRemovePagination(GenericCommandInteractionEvent)}
+ */
+public final class BookmarksPaginatorInteractor implements UserInteractor {
```

Comment:
before i CR this class, could u please give a detailed technical description / overview of the flow for the pagination in ur PR description?

just help me a bit to review this "monster", thanks

Reply:
thanks. i would appreciate though if u can give a quick rundown of the system overall in the PR. like how do the classes interact with each other? i saw some odd looking "request" and "accept" methods in the system etc.

im just a bit confused on whats going on overall with the pagination.

Reply:
I will add that to the description but an explanation for you now about the requestXXX: since the "backend" stuff is handled by BookmarksSystem it would make sense to tell the BookmarksSystem "Hey i have recieved a list command. Please request the paginator to display it", instead of passing all classes between each other. The only class all bookmark related classes know about is BookmarksSystem so if they want to interact with something they tell that the BookmarksSystem and noone else.

Reply:
Now has more extensive javadoc!